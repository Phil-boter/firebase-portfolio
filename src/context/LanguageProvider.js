import React, { createContext, useContext, useState } from "react";

export const LanguageContext = createContext(null);
