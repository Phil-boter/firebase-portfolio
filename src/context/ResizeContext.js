import { useEffect, useState, createContext } from "react";

export const ResizeContext = createContext();

function debounce(fn, ms) {
    let timer;
    return (cleanUp) => {
        clearTimeout(timer);
        timer = setTimeout((_) => {
            timer = null;
            fn.apply(this, arguments);
        }, ms);
    };
}

export function ResizeProvider({ children }) {
    const [dimensions, setDimensions] = useState({
        height: window.innerHeight,
        width: window.innerWidth,
    });

    console.log("dim in provider", dimensions);

    useEffect(() => {
        const debouncedHandleResize = debounce(function handleResize() {
            setDimensions({
                height: window.innerHeight,
                width: window.innerWidth,
            });
        }, 500);

        window.addEventListener("resize", debouncedHandleResize);

        return function cleanUp() {
            window.removeEventListener("resize", debouncedHandleResize);
        };
    }, [dimensions, setDimensions]);

    const value = { dimensions };

    return (
        <ResizeContext.Provider value={value}>
            {children}
        </ResizeContext.Provider>
    );
}
