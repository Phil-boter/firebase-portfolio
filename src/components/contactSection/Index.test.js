import React from "react";

import { render } from "@testing-library/react";

import { ResizeProvider } from "../../context/ResizeContext.js";
import { LanguageContext } from "../../context/LanguageProvider.js";
import ContactSection from "./Index.js";

let language = "en";

describe("MainPageAbout Component", () => {
    it("should render correctly and match snapshot", () => {
        const { container } = render(
            <ResizeProvider dimensions={{ height: 1000, widht: 1300 }}>
                <LanguageContext.Provider value={language}>
                    <ContactSection language={language} />
                </LanguageContext.Provider>
            </ResizeProvider>
        );
        expect(container).toMatchSnapshot();
    });

    it("should render ContactSection", () => {
        const { container } = render(
            <ResizeProvider dimensions={{ height: 1000, widht: 1300 }}>
                <LanguageContext.Provider value={language}>
                    <ContactSection language={language} />
                </LanguageContext.Provider>
            </ResizeProvider>
        );
        expect(container.querySelector(".section")).toBeInTheDocument();
    });
    it("should render Link, that redirect to GitHub if Click", () => {
        const clickToSpy = jest.fn();
        global.open = clickToSpy;
        const { container } = render(
            <ResizeProvider dimensions={{ height: 1000, widht: 1300 }}>
                <LanguageContext.Provider value={language}>
                    <ContactSection language={language} />
                </LanguageContext.Provider>
            </ResizeProvider>
        );
        expect(
            container.querySelector(".link-github").getAttribute("href")
        ).toBe("https://github.com/Phil-boter");

        expect(
            container.querySelector(".link-linkedIn").getAttribute("href")
        ).toBe("https://www.linkedin.com/in/philipp-dawid-759793206/");
    });
});
