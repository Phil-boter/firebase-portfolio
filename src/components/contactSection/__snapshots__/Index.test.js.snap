// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPageAbout Component should render correctly and match snapshot 1`] = `
<div>
  <section
    class="section"
  >
    <div
      class="section-image"
    >
      <img
        alt=""
        class="section-inner-image"
        src="/assets/denise-jans-p0QUpDUX8X8-unsplashsmall.jpg"
      />
    </div>
    <div
      class="section-article"
    >
      <div
        class="main-header contact-header header-before"
      >
        <h1>
          GET IN CONTACT
           
        </h1>
      </div>
      <article>
        <p>
          Find my code and my latest projects on
        </p>
         
        <span>
          <a
            class="link-github"
            data-testid="link-github"
            href="https://github.com/Phil-boter"
            rel="noreferrer"
            target="blank"
          >
            Github
          </a>
        </span>
         
        . 
      </article>
      <br />
      <article>
        <p>
          When you are interested in my educational background and professional experience visit my
        </p>
        <a
          class="link-linkedIn"
          data-testid="link-linkedin"
          href="https://www.linkedin.com/in/philipp-dawid-759793206/"
          rel="noreferrer"
          target="blank"
        >
          LinkedIn
        </a>
         profile or download my CV
         
        <a
          class="link-CV"
          href="/assets/CV-ENG.pdf"
          rel="noreferrer"
          target="blank"
          type="download"
        >
          here
        </a>
         or scan the QR-Code below.
      </article>
      <br />
      <article>
        <p>
          Interested in working with me? Feel free to
           
          <span
            class="link-mailto mail-link"
          >
            reach out.
          </span>
        </p>
         
      </article>
      <br />
      <article
        class="qr-code-container"
      >
        <img
          alt="qr-code to download cv"
          class="qr-code"
          src="/assets/qrcode_portfolio-philipp-dawid.web.app.png"
        />
      </article>
    </div>
  </section>
  <div
    class="email-modal"
  >
    <div
      class="email-main-wrapper"
    >
      <div
        class="contact-form-main-container"
      >
        <div
          class="contact-form-header main-header"
        >
          <h1>
            SCHREIBE MIR EINE EMAIL
          </h1>
        </div>
        <div
          class="contact-form-container"
          id="contact-form-container"
        >
          <div
            class="contact-form-inner-container"
            id="emailSlider"
          >
            <form
              class="contact-form"
              id="reset"
            >
              <label>
                Name
              </label>
              <input
                autocomplete="on"
                class="contact-form-input"
                id="imput-field-name"
                name="name"
                placeholder="Name"
                required=""
                type="text"
                value=""
              />
              <label>
                Email
              </label>
              <input
                autocomplete="on"
                class="contact-form-input"
                id="imput-field-email"
                name="email"
                placeholder="Email"
                required=""
                type="email"
                value=""
              />
              <label>
                Title
              </label>
              <input
                class="contact-form-input"
                id="input-field-title"
                name="Title"
                placeholder="Titel"
                required=""
                type="text"
                value=""
              />
              <label>
                Nachricht
              </label>
              <textarea
                class="contact-form-input contact-form-textarea"
                id="input-textarea"
                name="message"
                placeholder="Schreibe deine Nachricht hier"
                required=""
                rows="9"
              />
              <button
                class="button-contact-form send"
              >
                Absenden
              </button>
              <button
                class="button-contact-form"
              >
                Reset
              </button>
              <label
                class="ohnohoney"
              />
              <input
                autocomplete="off"
                class="ohnohoney"
                id="input-name-honeypot"
                name="name"
                placeholder="Your name here"
                type="text"
              />
              <label
                class="ohnohoney"
              />
              <input
                autocomplete="off"
                class="ohnohoney"
                id="input-email-honeypot"
                name="email"
                placeholder="Your e-mail here"
                type="emely"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
